Sockets/BackEndMonitoringSockets/src/CAcceptorWrapper.cpp

```diff
@@ -1,36 +1,45 @@
 #include "stdafx.h"
 #include "CAcceptorWrapper.h"
+#include "DataReceiver.h"
+#include "CServiceConnectionHandler.h"
+#include "CEvent.h"
 
 CAcceptorWrapper::CAcceptorWrapper(int port, const std::string& ip_address, 
-	size_t num_threads, CEvent& event, std::shared_ptr<CLogger> logger)
-	: m_event(event), m_logger(logger)
+	size_t num_threads, CEvent& event)
+	: m_event(event)
 {
 	m_server_acceptor = InitAcceptor(1111, "127.0.0.1");
-	m_stream = InitSocketWrapper(m_server_acceptor->GetHandle());
-	m_service_handler = InitServiceHandler(m_server_acceptor->GetHandle());
+	m_stream = InitSocketWrapper();
+	m_service_handler = InitServiceHandler();
 	m_pool = InitThreadPool(num_threads, m_event);
 }
 
 void CAcceptorWrapper::StartServer()
 {
 	std::cout << "Start server" << std::endl;
 	int handle = SOCKET_ERROR;
-	while (true)
+	while (!m_event.WaitFor(std::chrono::nanoseconds(1)))
```

Comment:
What is the point of this !m_event.WaitFor(std::chrono::nanoseconds(1) ??

Reply:
It's a temporary solution to check event state, it'll be changed when Check() method will be added to CEvent